mipengine/controller/api/DTOs/AlgorithmRequestDTO.py

```diff
@@ -1,14 +1,26 @@
 from dataclasses import dataclass
+from dataclasses import field
 from typing import Any
 from typing import Dict
+from typing import List
 from typing import Optional
 
 from dataclasses_json import dataclass_json
 
 
+@dataclass_json
+@dataclass
+class AlgorithmInputDataDTO:
+    pathology: str
+    dataset: List[str]
```

Comment:
dataset->datasets

Reply:
@jassak In the MIP-Engine interface it was decided to use dataset.

Should we use dataset or datasets?

I also agree with the datasets approach. With have a group of sets of data => group of dataset => datasets

The same goes for filter. Should we use filter or filters?

What do you think?

Reply:
since more than one datasets can be passed why it was decided to use dataset? And the same goes for the filters. And anything else that can be more than one?

Reply:
@jassak ;;;

Reply:
> In the MIP-Engine interface it was decided to use dataset.

When was this decided?

Ok, so in the database the field is called dataset since each row belongs to one and only one dataset. Likewise, gender is not called genders for example. However, as a parameter to some algorithm, I agree to use datasets since it's a list. In general it's a good practice to name lists using plural.

Reply:
The filter however is not a list (correct?). It's an object (dictionary) that represents the recursive nature of the filter clauses. To me one filter can have multiple clauses, so no need for plural.

Reply:
You are right Jason, I wrote that on the interface, for some reason I thought it was you...

Yes, filter value is an object.

So, datasets and filter are the suggested names. @apmariglis , do you agree?

Reply:
datasets and filters was decided.